### PR TITLE
Add Resource Requirement Container to Deployment

### PR DESCRIPTION
**Pull request by dvaldivia**

### CODE REVIEW

File: operator-kustomize/deployment.yaml [hunk 1]
```diff
@@ -18,6 +18,9 @@ spec:
         - name: minio-operator
           image: minio/k8s-operator:v3.0.5
           imagePullPolicy: IfNotPresent
+          resources:
+            requests:
+              memory: 64Mi
```
Comment (harshavardhana) [inline operator-kustomize/deployment.yaml:23]: I would suggest choosing 256MiB to be safe @dvaldivia

Comment (kerneltime) [inline operator-kustomize/deployment.yaml:23]: Seems a bit stingy? The operator also maintains a cache locally of k8s API responses. Maybe 500megs or 1gig? Not sure how the k8s client sizes its cache.

Comment (dvaldivia) [reply operator-kustomize/deployment.yaml:23]: we see it using only 11Mb on average, but we can go to 256MiB, it would be loose

Comment (dvaldivia) [reply operator-kustomize/deployment.yaml:23]: We have a cache? I think ging to 256Mb like @harshavardhana suggest is a good balance

Comment (kerneltime) [reply operator-kustomize/deployment.yaml]: The k8s client does have caching internally, so far we have been testing with tiny k8s clusters and I am not sure how the memory foot print changes on larger clusters.

Comment (dvaldivia) [reply operator-kustomize/deployment.yaml]: We don't see Operator consume large amounts of memory, even at 256 seems too much memory to spare from the usual 11MB we see in development

**Pull request merged**